OOP/HW_1.py

```diff
@@ -0,0 +1,546 @@
+# + Вывод на печать в удобном виде и возможность конвертации в строку
+# + Возможность проверить алфавит последовательности на корректность
+# 
+# Напишите класс `NucleicAcidSequence`:
+# + Данный класс реализует интерфейс `BiologicalSequence`
+# + Данный класс имеет новый метод `complement`, возвращающий комплементарную последовательность
+# + Данный класс имеет новый метод `gc_content`, возвращающий GC-состав (без разницы, в процентах или в долях)
+# 
+# Напишите классы наследники `NucleicAcidSequence`: `DNASequence` и `RNASequence`
+# + `DNASequence` должен иметь метод `transcribe`, возвращающий транскрибированную РНК-последовательность
+# + Данные классы не должны иметь <ins>публичных методов</ins> `complement` и метода для проверки алфавита, так как они уже должны быть реализованы в `NucleicAcidSequence`.
+# 
+# Напишите класс `AminoAcidSequence`:
+# + Данный класс реализует интерфейс `BiologicalSequence`
+# + Добавьте этому классу один любой метод, подходящий по смыслу к аминокислотной последовательности. Например, метод для нахождения изоэлектрической точки, молекулярного веса и т.д.
+# 
+# Комментарий по поводу метода `NucleicAcidSequence.complement`, так как я хочу, чтобы вы сделали его опредедённым образом:
+# 
+# При вызове `dna.complement()` или условного `dna.check_alphabet()` должны будут вызываться соответствующие методы из `NucleicAcidSequence`. При этом, данный метод должен обладать свойством полиморфизма, иначе говоря, внутри `complement` не надо делать условия а-ля `if seuqence_type == "DNA": return self.complement_dna()`, это крайне не гибко. Данный метод должен опираться на какой-то общий интерфейс между ДНК и РНК. Создание экземпляров `NucleicAcidSequence` не подразумевается, поэтому код `NucleicAcidSequence("ATGC").complement()` не обязан работать, а в идеале должен кидать исключение `NotImplementedError` при вызове от экземпляра `NucleicAcidSequence`
+# 
+# Вся сложность задания в том, чтобы правильно организовать код. Если у вас есть повторяющийся код в сестринских классах или родительском и дочернем, значит вы что-то делаете не так.
+# 
+# 
+# Маленькое замечание: По-хорошему, между классом `BiologicalSequence` и классами `NucleicAcidSequence` и `AminoAcidSequence`, ещё должен быть класс-прослойка, частично реализующий интерфейс `BiologicalSequence`, но его писать не обязательно, так как задание и так довольно большое (правда из-за этого у вас неминуемо возникнет повторяющийся код в классах `NucleicAcidSequence` и `AminoAcidSequence`)
+
+# In[36]:
+
+
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__(self):
+        pass
+    
+    @abstractmethod
+    def __getitem__(self, slc):
+        pass
+    
+    @abstractmethod
+    def __repr__(self):
+        pass
+    
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+    
+
+class BiologicalSequence_impl(BiologicalSequence):
```

Comment:
Классы в питоне назваются только при помощи CamelCase, за исключением встроенных
```suggestion
class BiologicalSequenceImpl(BiologicalSequence):
```